dll/win32/iphlpapi/iphlpapi_main.c

```diff
@@ -2620,12 +2621,113 @@ SetIpForwardEntryToStack(PMIB_IPFORWARDROW pRoute)
 }
 
 DWORD WINAPI
-NhGetInterfaceNameFromDeviceGuid(DWORD dwUnknown1,
-                                 DWORD dwUnknown2,
-                                 DWORD dwUnknown3,
+NhGetInterfaceNameFromDeviceGuid(GUID * pInterfaceGUID,
+                                 PWCHAR pInterfaceName,
+                                 PULONG pOutBufLen,
                                  DWORD dwUnknown4,
                                  DWORD dwUnknown5)
 {
-    FIXME("NhGetInterfaceNameFromDeviceGuid() stub\n");
-    return 0L;
+    UNICODE_STRING GuidString;
+    WCHAR szKeyName[2*MAX_PATH];
+    HKEY hKey;
+    DWORD result, type;
+
+    if (pInterfaceGUID == NULL || pOutBufLen == NULL)
+    {
+        return ERROR_INVALID_PARAMETER;
+    }
+
+    if (RtlStringFromGUID(pInterfaceGUID, &GuidString) != STATUS_SUCCESS)
+    {
+        // failed to convert guid to string
+        return ERROR_INVALID_PARAMETER;
+    }
+
+    if (StringCbPrintfW(szKeyName, sizeof(szKeyName), L"SYSTEM\\CurrentControlSet\\Control\\Network\\{4D36E972-E325-11CE-BFC1-08002BE10318}\\%s\\Connection", GuidString.Buffer) != S_OK)
+    {
+        // key name is too long
+        RtlFreeUnicodeString(&GuidString);
+        return ERROR_BUFFER_OVERFLOW;
+    }
+    RtlFreeUnicodeString(&GuidString);
+
+    if (RegOpenKeyExW(HKEY_LOCAL_MACHINE, szKeyName, 0, KEY_READ, &hKey) != ERROR_SUCCESS) 
+    {
+        // failed to find adapter entry
+        return ERROR_FILE_NOT_FOUND;
+    }
+
+    result = RegQueryValueExW(hKey, L"Name", NULL, &type, (PVOID)pInterfaceName, pOutBufLen);
+
+    if (result == ERROR_MORE_DATA)
+    {
+        RegCloseKey(hKey);
+        return result;
+    }
+
+    if (result != ERROR_SUCCESS || type != REG_SZ)
+    {
+        // failed to read adapter name
+        RegCloseKey(hKey);
+        return ERROR_NO_DATA;
+    }
+
+    RegCloseKey(hKey);
+    return ERROR_SUCCESS;
+}
+
+DWORD WINAPI
+NhGetInterfaceNameFromGuid(GUID * pInterfaceGUID,
```

Comment:
Same comments as previous function

Reply:
Actually, your implementation of those two functions seems to be totally identical? If that's correct, just call one from the other. Otherwise, you've got a bug ;p

Reply:
> If that's correct, just call one from the other.

Well, I don't know the differences between these two functions, but I doubt it's correct to call one from another. I used API Monitor to trace parameters, and it did not showed that one function calls another in any way.